src/bridge/graphs/peg_out.rs

```diff
@@ -335,155 +429,232 @@ impl PegOutGraph {
             Input {
                 outpoint: OutPoint {
                     txid: peg_in_confirm_txid,
-                    vout: take2_vout0.to_u32().unwrap(),
+                    vout: take_2_vout_0.to_u32().unwrap(),
+                },
+                amount: self.take_2_transaction.prev_outs()[take_2_vout_0].value, // Self-referencing
+            },
+            Input {
+                outpoint: OutPoint {
+                    txid: assert_txid,
+                    vout: take_2_vout_1.to_u32().unwrap(),
                 },
-                amount: self.take2_transaction.prev_outs()[0].value, // Self-referencing
+                amount: assert_transaction.tx().output[take_2_vout_1].value,
             },
             Input {
                 outpoint: OutPoint {
                     txid: assert_txid,
-                    vout: take2_vout1.to_u32().unwrap(),
+                    vout: take_2_vout_2.to_u32().unwrap(),
                 },
-                amount: assert_transaction.tx().output[take2_vout1].value,
+                amount: assert_transaction.tx().output[take_2_vout_2].value,
             },
             Input {
                 outpoint: OutPoint {
                     txid: assert_txid,
-                    vout: take2_vout2.to_u32().unwrap(),
+                    vout: take_2_vout_3.to_u32().unwrap(),
                 },
-                amount: assert_transaction.tx().output[take2_vout2].value,
+                amount: assert_transaction.tx().output[take_2_vout_3].value,
             },
         );
 
         let script_index = 1; // TODO replace placeholder
-        let disprove_vout0 = 1;
-        let disprove_vout1 = 2;
+        let disprove_vout_0 = 1;
+        let disprove_vout_1 = 2;
         let disprove_transaction = DisproveTransaction::new_for_validation(
             self.network,
+            &self.operator_taproot_public_key,
             &self.n_of_n_taproot_public_key,
             Input {
                 outpoint: OutPoint {
                     txid: assert_txid,
-                    vout: disprove_vout0.to_u32().unwrap(),
+                    vout: disprove_vout_0.to_u32().unwrap(),
                 },
-                amount: assert_transaction.tx().output[disprove_vout0].value,
+                amount: assert_transaction.tx().output[disprove_vout_0].value,
             },
             Input {
                 outpoint: OutPoint {
                     txid: assert_txid,
-                    vout: disprove_vout1.to_u32().unwrap(),
+                    vout: disprove_vout_1.to_u32().unwrap(),
                 },
-                amount: assert_transaction.tx().output[disprove_vout1].value,
+                amount: assert_transaction.tx().output[disprove_vout_1].value,
             },
             script_index,
         );
 
-        let burn_vout0 = 2;
-        let burn_transaction = BurnTransaction::new_for_validation(
+        let disprove_chain_vout_0 = 2;
```

Comment:
Should be kick_off_2 - output 1
```suggestion
        let disprove_chain_vout_0 = 1;
```